### PR TITLE
Add flag to cache libcc downloads

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Will need a follow up PR to electron to allow usage of this, basically you can optionally cache libcc downloads to a folder on your machine.  Means that when you switch branch you won't have to redownload stuff.

### CODE REVIEW

**Comment by alexeykuzmin** [comment]
Would it make sense to give unique local names to already downloaded libcc archives to avoid redownload?

**Comment by MarshallOfSound** [comment]
> Would it make sense to give unique local names to already downloaded libcc archives to avoid redownload?

What do you mean, all libcc downloads are stored at `{commit}-{filename}` so they won't redownload 🤔

**Comment by alexeykuzmin** [comment]
> What do you mean, all libcc downloads are stored at `{commit}-{filename}` so they won't redownload 🤔

@MarshallOfSound 
Sorry, I didn't read your changes as carefully as I should have.
That's exactly what you're doing.

File: script/download [hunk 1]
```diff
@@ -162,6 +189,13 @@ def generate_filenames_gypi(target_file, libcc_source_path,
                          libcc_source_path, libcc_shared_library_path,
                          libcc_static_library_path])
 
+def mkdir_p(path):
```
Comment (alexeykuzmin) [inline script/download]: Already exists in `lib/filesystem.py`.

File: script/download [hunk 2]
```diff
@@ -9,11 +9,15 @@ import sys
 import tempfile
 import urllib2
 import tarfile
+import shutil
```
Comment (alexeykuzmin) [inline script/download]: Imports should be listed in the alphabetical order.

File: script/download [hunk 3]
```diff
@@ -80,6 +87,7 @@ def parse_args():
                       help='Overwrite destination if it already exists.')
   parser.add_argument('-c', '--commit', nargs='?', default='HEAD',
                       help='The commit of libchromiumcontent to download.')
+  parser.add_argument('--cache', action='store_true')
```
Comment (alexeykuzmin) [inline script/download]: `--use_cache` maybe? `cache` is slightly weird name for a boolean argument.

File: script/download [hunk 4]
```diff
@@ -124,30 +132,49 @@ def download_if_needed(destination, base_url, commit, filename, force):
...
 
-  download(destination, base_url, commit, filename)
+  download(destination, base_url, commit, filename, use_cache)
 
   with open(version_file, 'w') as f:
     f.write('{0}\n'.format(commit))
 
 
-def download(destination, base_url, commit, filename):
+def download(destination, base_url, commit, filename, use_cache):
```
Comment (alexeykuzmin) [inline script/download]: I would probably move the whole logic here. Like
```python
archive = None
cache_key = ...
url = ...

if use_cache:
  archive = get_from_cache(cache_key)

if archive is None:
  archive = download(url)
  if archive is not None and use_cache:
    save_to_cache(key=cache_key, file=archive)

if archive is not None:
  extract(from=archive, to=destination)
else:
  print 'Better luck next time, mate'
```

**Pull request closed without merging**